code/modules/pai/card.dm

```diff
@@ -268,6 +269,7 @@
 	if(pai)
 		return FALSE
 	pai = downloaded
+	RegisterSignal(pai, COMSIG_QDELETING, PROC_REF(on_pai_del))
```

Comment:
Doesn't this need to unregister?

Reply:
I don't think so, since destroy() will already unregister it along with all other signals, and the pai mob shouldn't exist without a pai card under normal circumstances.

Reply:
The pAI mob's destroy won't unregister this wouldn't it? I haven't looked at it recently but I thought the pAI card would unregister here, which can exist without the mob

Reply:
Any destroyed datum will unregister signals registered by or on it. This behavior can be found in `datum/Destroy`.

Reply:
u right